### PR TITLE
Changed devcontainer to use smaller image

### PR DESCRIPTION
**Pull request by keyluck**
# Proposed Changes

Removed ~8gbs worth of tools we don't use from the default codespaces image to speed up the rebuilding process for our codespaces. I pushed the image to our github packages and added secrets to our codespace environment to [pull the image from a private repository](https://docs.github.com/en/codespaces/codespaces-reference/allowing-your-codespace-to-access-a-private-image-registry). I kept the install scripts for helm and go even though we don't use them since we most likely will in the future.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

- [x] My changes generate no new warnings
- [x] Logging has been added where applicable (Error handling, user-driven events, API requests/responses)

- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [x] Tests meet the test coverage threshold

## Reviewer Checklist

- [ ] Is the code easy to understand?
- [ ] Are the changes well documented?

- [ ] Can you think of any use case in which the code does not behave as intended?
- [ ] Can you think of any inputs or external events that could break the code?

- [ ] Should any logging or debugging information be added or removed?

## Further Comments

If this is a relatively large or complex change, kick off the discussion by explaining how you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by rianfowler — COMMENTED** [review]
- What secrets did you add to the local environment? 
- how did you scope the access? 
- Can document the names of the secrets the codespaces requires?

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,51 +1,57 @@
-// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
-// https://github.com/microsoft/vscode-dev-containers/tree/v0.177.0/containers/javascript-node-postgres
-// Update the VARIANT arg in docker-compose.yml to pick a Node.js version: 10, 12, 14
 {
-  "name": "Node.js & PostgreSQL (Community)",
-  "dockerComposeFile": "docker-compose.yml",
-  "service": "app",
-  "workspaceFolder": "/workspace",
+	"name": "Docker in Docker",
+	//"dockerFile": "Dockerfile",
```
Comment (KaemonIsland) [inline .devcontainer/devcontainer.json:3]: Should this be removed?

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -1,51 +1,57 @@
...
+	"dockerComposeFile": "docker-compose.yml",
+  	"service": "app",
+  	"workspaceFolder": "/workspace",
+	//"mounts": ["source=dind-var-lib-docker,target=/var/lib/docker,type=volume"],
```
Comment (KaemonIsland) [inline .devcontainer/devcontainer.json:7]: Same here

**Review by KaemonIsland — APPROVED** [review]

File: .talismanrc [hunk 3]
```diff
@@ -1,3 +1,7 @@
 fileignoreconfig:
 - filename: app-config.yaml
-  checksum: 70e54692b5d28f0bb25d0c4beac3401c28004a1866150ed7f08dd7eeeb0cb63b
\ No newline at end of file
+  checksum: 70e54692b5d28f0bb25d0c4beac3401c28004a1866150ed7f08dd7eeeb0cb63b
```
Comment (rianfowler) [inline .talismanrc:3]: how did this get updated?

File: docs/codespace.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+<*>_CONTAINER_REGISTRY_USER
+<*>_CONTAINER_REGISTRY_PASSWORD
+```
+
+You can generate a personal access token with the `read:packages` permission to create these secrets for Codespaces:
+
+```
+DEV_CONTAINER_REGISTRY_SERVER = ghcr.io
+DEV_CONTAINER_REGISTRY_USER = <username>
+DEV_CONTAINER_REGISTRY_PASSWORD = <PAT>
```
Comment (rianfowler) [inline docs/codespace.md:26]: Do these environment variables need to be stored on our account level code spaces secrets? The codespace works for me without doing this configuration.

**Review by rianfowler — COMMENTED** [review]
This change causes CORS errors when running `yarn dev` 

```
Access to XMLHttpRequest at 'http://127.0.0.1:7000/api/catalog/entities?fields=kind' from origin 'http://127.0.0.1:64781' has been blocked by CORS policy: The 'Access-Control-Allow-Origin' header has a value 'http://127.0.0.1:3000' that is not equal to the supplied origin.
```

We'll need to update the backend config to return an `Allow-Origin` header with the new port. LMK if you need help with this.

File: docs/codespace.md (hunk 4 above)
Comment (keyluck) [reply docs/codespace.md:26]: I did it at the repository level, I did check inside codespaces and the value appears to be encrypted. I removed them from the repository level anyways so we can each use our own.

File: .talismanrc (hunk 3 above)
Comment (rianfowler) [reply .talismanrc:3]: Would it be okay if I added a task to isolate the parts that talisman is complaining about? I think ignoring the whole app config is a bad idea long term.

**Review by rianfowler — APPROVED** [review]

**Pull request merged**